src/runtime/importhook.cs

```diff
@@ -112,13 +129,26 @@ internal static void Initialize()
         /// </summary>
         internal static void Shutdown()
         {
-            if (Runtime.Py_IsInitialized() != 0)
+            if (Runtime.Py_IsInitialized() == 0)
             {
-                RestoreImport();
+                return;
+            }
 
-                Runtime.XDecref(py_clr_module);
-                Runtime.XDecref(root.pyHandle);
+            RestoreImport();
+
+            bool shouldFreeDef = Runtime.Refcount(py_clr_module) == 1;
+            Runtime.XDecref(py_clr_module);
+            py_clr_module = IntPtr.Zero;
+#if PYTHON3
```

Comment:
Can we avoid introducing another compile-time check, and use version info from `Runtime` class instead?

Reply:
Actually, in the modernisation branch I gave up on the idea of having Python 2 and 3 in the same codebase, especially considering the imminent EoL of Python 2. I do that by defaulting to Python 3 and letting the user supply `PYTHON2` explicitly for the Py2 version. So here, to stay in line with that, it would be enough to change to `#if !PYTHON2`.